diff --git a/tesseract.pxd b/tesseract.pxd
index 4c8eea9..4ab0a4d 100644
--- a/tesseract.pxd
+++ b/tesseract.pxd
@@ -53,6 +53,12 @@ cdef extern from "leptonica/allheaders.h" nogil:
         L_SEVERITY_ERROR    = 5   # Print error and higher messages
         L_SEVERITY_NONE     = 6   # Highest severity: print no messages
 
+cdef extern from "leptonica/allheaders.h" nogil:
+    pass
+
+cdef extern from "leptonica/pix_internal.h" nogil:
+    struct Pix:
+
 cdef extern from "tesseract/publictypes.h" nogil:
     cdef enum PolyBlockType:
         PT_UNKNOWN          # Type is not yet known. Keep as the first element.
