diff --git a/tests/test_api.py b/tests/test_api.py
index 766f59b..48fed97 100644
--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -183,7 +183,7 @@ class TestTessBaseApi(unittest.TestCase):
         self.assertEqual([v[0] for v in mapped_confidences], words)
         self.assertEqual([v[1] for v in mapped_confidences], confidences)
 
-    @unittest.skipIf(_TESSERACT_VERSION < 0x4000000, "tesseract < 4")
+    @unittest.skipIf(_TESSERACT_VERSION < 0x4000000 or _TESSERACT_VERSION >= 0x5000000, "tesseract < 4")
     def test_LSTM_choices(self):
         """Test GetBestLSTMSymbolChoices."""
         self._api.SetVariable("lstm_choice_mode", "2")
